cloudberry/zion/src/main/scala/edu/uci/ics/cloudberry/zion/model/impl/SQLGenerator.scala

```diff
@@ -0,0 +1,601 @@
+    filter.relation match {
+      case Relation.matches => {
+        val values = filter.values.map(_.asInstanceOf[String])
+        s"""$fieldExpr="${values(0)}""""
+      }
+      case Relation.!= => {
+        val values = filter.values.map(_.asInstanceOf[String])
+        s"""$fieldExpr!="${values(0)}""""
+      }
+      case Relation.contains => {
+        val values = filter.values.map(_.asInstanceOf[String])
+        s"lower($fieldExpr) ${stringContains} '%${values(0)}%'"
+      }
+    }
+  }
+
+  protected def parseTextRelation(filter: FilterStatement, fieldExpr: String): String = {
+    val wordsArr = ArrayBuffer[String]()
+    filter.values.foreach(w => wordsArr += w.toString)
+    val sb = new StringBuilder(s"${fullTextMatch(0)}($fieldExpr) ${fullTextMatch(1)} ('")
+    sb.append(wordsArr.mkString("+"," +","") + s"' in boolean mode)")
+    sb.toString()
+  }
+
+  protected def parseNumberRelation(filter: FilterStatement,
+                                    fieldExpr: String): String = {
+    filter.relation match {
+      case Relation.inRange =>
+        if (filter.values.size != 2) throw new QueryParsingException(s"relation: ${filter.relation} require two parameters")
+        s"$fieldExpr >= ${filter.values(0)} and $fieldExpr < ${filter.values(1)}"
+      case Relation.in =>
+        if (!filter.values.isEmpty) {
+          s"$fieldExpr in ( ${filter.values.mkString(",")} )"
+        } else {
+          s"true"
+        }
+      case _ =>
+        s"$fieldExpr ${filter.relation} ${filter.values(0)}"
+    }
+  }
+
+  protected def parseAggregateFunc(aggr: AggregateStatement,
+                                   fieldExpr: String): String = {
+    def aggFuncExpr(aggFunc: String): String = {
+      if (aggr.field.name.equals("*")) {
+        s"$aggFunc($groupVar)"
+      } else {
+        s"$aggFunc($fieldExpr)"
+      }
+    }
+
+    aggr.func match {
+      case topK: TopK => ???
+      case DistinctCount => ???
+      case _ =>
+        aggFuncExpr(getAggregateStr(aggr.func))
+    }
+  }
+
+  private def parseFilter(filters: Seq[FilterStatement], exprMap: Map[String, FieldExpr], unnestTestStrs: Seq[String], queryBuilder: StringBuilder): ParsedResult = {
+    if (filters.isEmpty && unnestTestStrs.isEmpty) {
+      ParsedResult(Seq.empty, exprMap)
+    } else {
+      val filterStrs = filters.map { filter =>
+        filter.field.dataType match {
+          case _ =>
+            parseFilterRelation(filter, exprMap(filter.field.name).refExpr)
+        }
+      }
+      val filterStr = (unnestTestStrs ++ filterStrs).mkString("where ", " and ", "")
+      appendIfNotEmpty(queryBuilder, filterStr)
+      ParsedResult(Seq.empty, exprMap)
+    }
+  }
+
+  protected def parseGeoCell(scale: Integer, fieldExpr: String, dataType: DataType.Value): String = {
+    s"$geoAsText($dataType($truncate(${pointGetCoord(0)}($fieldExpr),$scale),$truncate(${pointGetCoord(1)}($fieldExpr),$scale))) "
```

Comment:
MySql ST_ASTEXT:
https://dev.mysql.com/doc/refman/5.7/en/gis-format-conversion-functions.html
MySql ST_X, ST_Y:
https://dev.mysql.com/doc/refman/5.6/en/gis-point-property-functions.html
MySql truncate:
http://www.w3resource.com/mysql/mathematical-functions/mysql-truncate-function.php
The other functions are mostly the same as other DB.